superset-frontend/src/explore/components/controls/DatasourceControl/index.jsx

```diff
@@ -267,11 +267,12 @@ class DatasourceControl extends React.PureComponent {
       showSaveDatasetModal,
     } = this.state;
     const { datasource, onChange, theme } = this.props;
-    const isMissingDatasource = datasource?.id == null;
+    const isMissingDatasource = !datasource || datasource.id === 0;
```

Comment:
Would this also work? 
```suggestion
    const isMissingDatasource = !!datasource?.id
```

Reply:
That's cleaner, thanks!  This now also covers the original `null` case in case it's still used somewhere I couldn't find, which is probably good.

Reply:
Actually sorry, `!!datasource?.id` doesn't work but `!datasource?.id` works.  Is that what you meant?

Reply:
yeah, that's it. 👍